training_data_loader.cpp

```diff
@@ -490,6 +490,57 @@ struct FeaturedBatchStream : Stream<StorageT>
 
 extern "C" {
 
+    EXPORT SparseBatch* get_sparse_batch_from_fens(
+        const char* feature_set_c,
+        int num_fens,
+        const char* const* fens,
+        int* scores,
+        int* plies,
+        int* results
+    )
+    {
+        std::vector<TrainingDataEntry> entries;
+        entries.reserve(num_fens);
+        for (int i = 0; i < num_fens; ++i)
+        {
+            TrainingDataEntry e;
+            e.pos = Position::fromFen(fens[i]);
+            movegen::forEachLegalMove(e.pos, [&](Move m){e.move = m;});
+            e.score = scores[i];
+            e.ply = plies[i];
+            e.result = results[i];
+            entries.push_back(std::move(e));
```

Comment:
emplace_back ?

Reply:
Could use emplace_back() and then use entries.back(), but this is not a performance critical part.

Reply:
Okay, I guess it's simpler like that. Modified.

Reply:
thanks!